dts/arm/renesas/ra/ra-cm4-common.dtsi

```diff
@@ -115,6 +116,135 @@
 			};
 		};
 
+		ioport0: gpio@40040000 {
+			compatible = "renesas,ra-gpio";
+			reg = <0x40040000 0x20>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			port-no = <0>;
+			ngpios = <16>;
+			interrupts = <RA_ICU_IRQ_UNSPECIFIED 0 RA_ICU_PORT_IRQ2>,
+				     <RA_ICU_IRQ_UNSPECIFIED 0 RA_ICU_PORT_IRQ3>,
+				     <RA_ICU_IRQ_UNSPECIFIED 0 RA_ICU_PORT_IRQ6>,
+				     <RA_ICU_IRQ_UNSPECIFIED 0 RA_ICU_PORT_IRQ7>,
+				     <RA_ICU_IRQ_UNSPECIFIED 0 RA_ICU_PORT_IRQ10>,
+				     <RA_ICU_IRQ_UNSPECIFIED 0 RA_ICU_PORT_IRQ15>;
+			interrupt-names = "port-irq2", "port-irq3", "port-irq6",
+					  "port-irq7", "port-irq10", "port-irq15";
+			port-irq2-pins = <2>;
+			port-irq3-pins = <4>;
+			port-irq6-pins = <0>;
+			port-irq7-pins = <1 15>;
+			port-irq10-pins = <5>;
+			port-irq15-pins = <11>;
+			status = "disabled";
+		};
+
+		ioport1: gpio@40040020 {
+			compatible = "renesas,ra-gpio";
+			reg = <0x40040020 0x20>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			port-no = <1>;
```

Comment:
why is this property needed? this is not hw description?

Reply:
Yes, it is an index of gpio.
The index can be calculated from the address.
I replaced it with the calculated value.

Reply:
fancy, you sure the new one is a good idea? it relies on the port memory space being contiguous (question for both of you @soburi @gmarull)

Reply:
In the RA series, all GPIO ports use continuous addresses. Even if there is a dropout in the middle, the port number will not change, so calculation is possible.

CM4-based SoC uses 0x40040000 to 0x40040160 for ioport0 .. ioportb.
In the case of CM33-based SoC, it also uses 0x40080000 to 0x40080160.

Reply:
In past situations, we've always restricted DT to the hardware description itself (port-no is needed here because of how software works, not because it describes hw uniquely).